### PR TITLE
Auction website monitoring

### PR DESCRIPTION
**Pull request by weilbith**
Close: trustlines-network/deployment-config#36

I checked the functionality locally with the dev-server of the foundation webpage. Changing the code (which gets automatically reloaded) caused an error warning.

@schmir-at-bb I already added the parts at the `CHANGELOG`. The intention was to save work from you. Feel free to adjust for a better version number.

### CODE REVIEW

File: tlwatch/auction_website.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    for script_url in script_url_list:
+        script = requests.get(script_url).text
+        source_list.append(script)
+
+    source_hash = md5("".join(source_list).encode())
+    return source_hash.hexdigest()
+
+
+def watch_auction_website(url: str, original_hash: str):
+    current_hash = calculate_website_source_hash(url)
```
Comment (schmir-at-bb) [inline tlwatch/auction_website.py]: Please catch errors from requests here and report them as error to riemann. You can report with status "io-error" and report more details as "description" field.

**Comment by weilbith** [comment]
@schmir-at-bb thanks for the review! You have some really important point I hadn't in mind on developing. Will start working on them. :+1: 
Regarding the JS sources: I get your point of the too specific `js` directory prefix. But would you agree that only relative sources should be fetched? This also keeps the code simple, since I don't have to differ on the host extension. So I would not include `jquery` etc.

**Comment by weilbith** [comment]
I just realized that such distinction is hard to archive. Some sources also just missing the protocol prefix... So I get just rid of the filter. Results into possible false positives where we don't have a clue what happened. But I guess this is very unlikely to happen.

File: tlwatch/auction_website.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    script_url_list = [
+        parse.urljoin(url, script.get("src"))
+        for script in soup.findAll("script", attrs={"src": re.compile("^/js")})
+    ]
+
+    for script_url in script_url_list:
+        script = requests.get(script_url).text
+        source_list.append(script)
+
+    source_hash = md5("".join(source_list).encode())
```
[the remark this answers is not in the document]
Comment (weilbith) [reply tlwatch/auction_website.py]: :heavy_check_mark:

File: tlwatch/auction_website.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    return source_hash.hexdigest()
+
+
+def watch_auction_website(url: str, original_hash: str):
+    current_hash = calculate_website_source_hash(url)
+    state = "ok" if current_hash == original_hash else "error"
+    return [{"service": "auction-website.source-hash", "state": state, "ttl": 30}]
```
[the remark this answers is not in the document]
Comment (weilbith) [reply tlwatch/auction_website.py]: :heavy_check_mark:

File: tlwatch/auction_website.py (hunk 1 above)
Comment (weilbith) [reply tlwatch/auction_website.py]: I've took you're error handling here as example. :heavy_check_mark:

**Comment by weilbith** [comment]
So the only left point from your list is the idea about the JSON file. So you would like to get rid of the `<script.../>` extradition, but fetch all links in the JSON and check them individually. So if the HTML is not correct anymore, this would cause its own error report, as if any JS script is "broken"?

**Comment by schmir-at-bb** [comment]
Let's try if what we have now is already good enough.

**Pull request merged**